apps/explorer/src/pages/AddressPage/AddressPage.tsx

```diff
@@ -28,10 +30,16 @@ const AddressPage = ({ addressStr }: AddressPageProps) => {
   const { t } = useTranslation()
   const theme = useTheme()
   const dispatch = useAppDispatch()
+  const navigate = useNavigate()
 
   useEffect(() => {
+    if (!isValidAddress(addressStr)) {
+      navigate('/404')
+      return
+    }
```

Comment:
I would put this not in this component but in the one that instantiates it in the existing validation without the need of a useEffect, as I wrote in the previous comment,

Reply:
Done